File: CS/SnapObjectDataSourceExample/Form1.cs
using DevExpress.Snap.Services;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.DataAccess.ObjectBinding;

namespace SnapObjectDataSourceExample {
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm {
        IObjectDataSourceValidationService defaultValidationService;

        public Form1() {
            InitializeComponent();

            #region #ObjectDataSourceLoadingBehavior
            snapControl1.Options.DataSourceOptions.ObjectDataSourceLoadingBehavior = DevExpress.DataAccess.DocumentLoadingBehavior.Prompt;
            #endregion #ObjectDataSourceLoadingBehavior

            defaultValidationService = snapControl1.GetService<IObjectDataSourceValidationService>();

            repositoryItemDocumentLoadingBehavior.Items.AddRange(typeof(DevExpress.DataAccess.DocumentLoadingBehavior).GetEnumValues());
            barComboDocumentLoadingBehavior.EditValue = snapControl1.Options.DataSourceOptions.ObjectDataSourceLoadingBehavior;

            this.barComboDocumentLoadingBehavior.EditValueChanged += new System.EventHandler(this.barComboDocumentLoadingBehavior_EditValueChanged);
            this.barCheckCustomValidationService.CheckedChanged += BarCheckCustomValidationService_CheckedChanged;
        }

        private void BarCheckCustomValidationService_CheckedChanged(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            if (barCheckCustomValidationService.Checked) {
                #region #ReplaceService
                snapControl1.ReplaceService<IObjectDataSourceValidationService>(new MyObjectDataSourceValidationService());
                #endregion #ReplaceService
            }
            else {
                snapControl1.ReplaceService<IObjectDataSourceValidationService>(defaultValidationService);
            }
        }

        private void barBtnLoadSnx_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            snapControl1.LoadDocument("DataBoundReport.snx");
        }

        private void barComboDocumentLoadingBehavior_EditValueChanged(object sender, EventArgs e) {
            snapControl1.Options.DataSourceOptions.ObjectDataSourceLoadingBehavior = (DevExpress.DataAccess.DocumentLoadingBehavior)(sender as DevExpress.XtraBars.BarEditItem).EditValue;
        }
    }
    #region #MyObjectDataSourceValidationService
    public class MyObjectDataSourceValidationService : IObjectDataSourceValidationService {
        public void Validate(IEnumerable<ObjectDataSource> dataSources) {
            // Do nothing to allow loading.
            // Clear the DataSource and DataMember properties to prohibit loading.
            foreach (ObjectDataSource ds in dataSources) {
                ds.DataSource = null;
                ds.DataMember = null;
            };
        }
    }
    #endregion #MyObjectDataSourceValidationService
}

File: VB/SnapObjectDataSourceExample/Form1.vb
Imports DevExpress.Snap.Services
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.DataAccess.ObjectBinding

Namespace SnapObjectDataSourceExample
    Partial Public Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Private defaultValidationService As IObjectDataSourceValidationService

        Public Sub New()
            InitializeComponent()

'            #Region "#ObjectDataSourceLoadingBehavior"
            snapControl1.Options.DataSourceOptions.ObjectDataSourceLoadingBehavior = DevExpress.DataAccess.DocumentLoadingBehavior.Prompt
'            #End Region ' #ObjectDataSourceLoadingBehavior

            defaultValidationService = snapControl1.GetService(Of IObjectDataSourceValidationService)()

            repositoryItemDocumentLoadingBehavior.Items.AddRange(GetType(DevExpress.DataAccess.DocumentLoadingBehavior).GetEnumValues())
            barComboDocumentLoadingBehavior.EditValue = snapControl1.Options.DataSourceOptions.ObjectDataSourceLoadingBehavior

            AddHandler barComboDocumentLoadingBehavior.EditValueChanged, AddressOf barComboDocumentLoadingBehavior_EditValueChanged
            AddHandler Me.barCheckCustomValidationService.CheckedChanged, AddressOf BarCheckCustomValidationService_CheckedChanged
        End Sub

        Private Sub BarCheckCustomValidationService_CheckedChanged(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            If barCheckCustomValidationService.Checked Then
'                #Region "#ReplaceService"
                snapControl1.ReplaceService(Of IObjectDataSourceValidationService)(New MyObjectDataSourceValidationService())
'                #End Region ' #ReplaceService
            Else
                snapControl1.ReplaceService(Of IObjectDataSourceValidationService)(defaultValidationService)
            End If
        End Sub

        Private Sub barBtnLoadSnx_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barBtnLoadSnx.ItemClick
            snapControl1.LoadDocument("DataBoundReport.snx")
        End Sub

        Private Sub barComboDocumentLoadingBehavior_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            snapControl1.Options.DataSourceOptions.ObjectDataSourceLoadingBehavior = CType((TryCast(sender, DevExpress.XtraBars.BarEditItem)).EditValue, DevExpress.DataAccess.DocumentLoadingBehavior)
        End Sub
    End Class
    #Region "#MyObjectDataSourceValidationService"
    Public Class MyObjectDataSourceValidationService
        Implements IObjectDataSourceValidationService

        Public Sub Validate(ByVal dataSources As IEnumerable(Of ObjectDataSource)) Implements IObjectDataSourceValidationService.Validate
            ' Do nothing to allow loading.
            ' Clear the DataSource and DataMember properties to prohibit loading.
            For Each ds As ObjectDataSource In dataSources
                ds.DataSource = Nothing
                ds.DataMember = Nothing
            Next ds
        End Sub
    End Class
    #End Region ' #MyObjectDataSourceValidationService
End Namespace
